### PR TITLE
PE-3044: side bar, profile component, and feedbacks improvements

### PR DESCRIPTION
**Pull request by thiagocarvalhodev**
--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/7iaig3rvak148
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/5rfmjjnbpjoag

### CODE REVIEW

File: lib/authentication/ardrive_auth.dart [hunk 1]
```diff
@@ -14,20 +17,33 @@ abstract class ArDriveAuth {
   Future<bool> isUserLoggedIn();
   Future<bool> isExistingUser(Wallet wallet);
   Future<User> login(Wallet wallet, String password, ProfileType profileType);
+  Future<User> unlockWithBiometrics({required String localizedReason});
+
+  // Future<User> loginWithBiometrics(
```
Comment (karlprieb) [inline lib/authentication/ardrive_auth.dart:22]: ?

File: lib/authentication/ardrive_auth.dart [hunk 2]
```diff
@@ -114,12 +142,18 @@ class _ArDriveAuth implements ArDriveAuth {
...
       currentUser = await _userRepository.getUser(password);
 
+      logger.d('User unlocked');
+
       _userStreamController.add(_currentUser);
 
       return currentUser;
     } catch (e) {
+      logger.e(e.toString());
+      logger.e('Failed to unlock user with password', e);
```
Comment (karlprieb) [inline lib/authentication/ardrive_auth.dart:156]: if you're already loggin the `e` error, why logging it twice?
do `logger.e('Failed to unlock user with password', e);` or:
```
logger.e('Failed to unlock user with password');
logger.e(e.toString());
```

File: lib/utils/logger/logger.dart [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import 'package:logger/logger.dart';
+
+final logger = Logger();
```
Comment (karlprieb) [inline lib/utils/logger/logger.dart:3]: here we could set the log level based on the flavor.
We can set `warning` for production and `debug` for dev.

File: lib/authentication/login/views/login_page.dart [hunk 4]
```diff
@@ -304,15 +299,17 @@ class _PromptWalletViewState extends State<PromptWalletView> {
...
       child: _LoginCard(
         content: Column(
+          mainAxisSize: MainAxisSize.min,
           mainAxisAlignment: MainAxisAlignment.spaceEvenly,
           children: [
             Align(
               alignment: Alignment.topCenter,
               child: Text(
-                appLocalizationsOf(context).welcomeBack,
+                appLocalizationsOf(context).welcome,
```
Comment (karlprieb) [inline lib/authentication/login/views/login_page.dart:308]: @matibat we need to remove `welcomeBack` translation after we release this

File: lib/blocs/create_manifest/create_manifest_cubit.dart [hunk 5]
```diff
@@ -13,13 +12,12 @@ import 'package:collection/collection.dart';
...
 import 'package:flutter/widgets.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
-import 'package:reactive_forms/reactive_forms.dart';
 import 'package:uuid/uuid.dart';
 
 part 'create_manifest_state.dart';
 
 class CreateManifestCubit extends Cubit<CreateManifestState> {
-  late FormGroup form;
+  // late FormGroup form;
```
Comment (karlprieb) [inline lib/blocs/create_manifest/create_manifest_cubit.dart:20]: ?

File: lib/blocs/drive_attach/drive_attach_cubit.dart [hunk 6]
```diff
@@ -1,31 +1,33 @@
...
 import 'package:equatable/equatable.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 
 part 'drive_attach_state.dart';
 
 /// [DriveAttachCubit] includes logic for attaching drives to the user's profile.
 class DriveAttachCubit extends Cubit<DriveAttachState> {
-  late FormGroup form;
+  // late FormGroup form;
```
Comment (karlprieb) [inline lib/blocs/drive_attach/drive_attach_cubit.dart:19]: ?

File: lib/blocs/profile_unlock/profile_unlock_cubit.dart [hunk 7]
```diff
@@ -105,9 +105,9 @@ class ProfileUnlockCubit extends Cubit<ProfileUnlockState> {
       bool authenticated = false;
 
       if (needBiometrics) {
-        authenticated =
-            // ignore: use_build_context_synchronously
-            await _biometricAuthentication.authenticate(context);
+        // authenticated =
+        // ignore: use_build_context_synchronously
+        // await _biometricAuthentication.authenticate(context);
       }
```
Comment (karlprieb) [inline lib/blocs/profile_unlock/profile_unlock_cubit.dart:111]: ?

File: lib/components/create_manifest_form.dart [hunk 8]
```diff
@@ -346,102 +380,106 @@ class CreateManifestForm extends StatelessWidget {
...
-                                  ),
-                            ],
-                          ),
-                        ),
-                      ),
-                    ),
-                  ],
-                )),
+            content: _selectFolder(state, context),
+            //  SizedBox(
```
Comment (karlprieb) [inline lib/components/create_manifest_form.dart:413]: remove unused comment

File: lib/components/file_picker_modal.dart [hunk 9]
```diff
@@ -203,33 +205,28 @@ Future<bool> verifyStoragePermissionAndShowModalWhenDenied(
   return true;
 }
 
+// TODO: check this
```
Comment (karlprieb) [inline lib/components/file_picker_modal.dart:208]: what you need to check here?

File: lib/components/ghost_fixer_form.dart [hunk 10]
```diff
@@ -40,76 +41,79 @@ class GhostFixerForm extends StatelessWidget {
...
-                context,
-                driveId: state.viewingFolder.folder.driveId,
-                parentFolderId: state.viewingFolder.folder.id,
-              ),
-            );
-          } else {
-            return Container();
+          } else if (state is GhostFixerSuccess) {
+            Navigator.pop(context);
+            Navigator.pop(context);
```
Comment (karlprieb) [inline lib/components/ghost_fixer_form.dart:52]: why do we pop twice?

File: lib/components/side_bar.dart [hunk 11]
```diff
@@ -0,0 +1,577 @@
...
+        ),
+      ],
+    );
+  }
+
+  Widget _buildSideBarBottom() {
+    return _isExpanded
+        ? Padding(
+            padding: const EdgeInsets.only(
+              left: 43.0,
```
Comment (karlprieb) [inline lib/components/side_bar.dart:270]: why not `43`?

File: lib/pages/drive_detail/components/drive_detail_data_list.dart [hunk 12]
```diff
@@ -168,7 +168,7 @@ ArDriveDataTable _buildDataListContent(
   bool isMultiselecting,
 ) {
   return ArDriveDataTable<ArDriveDataTableItem>(
-    key: ValueKey(folder.id),
+    key: ValueKey(folder.id + items.length.toString()),
```
Comment (karlprieb) [inline lib/pages/drive_detail/components/drive_detail_data_list.dart:171]: why? `folder.id` is not unique enough?

File: lib/pages/drive_detail/components/drive_explorer_item_tile.dart [hunk 13]
```diff
@@ -343,19 +351,110 @@ class _DriveExplorerItemTileTrailingState
...
-    );
-  }
 }
 
 bool isMobile(BuildContext context) {
   final isPortrait = MediaQuery.of(context).orientation == Orientation.portrait;
   return isPortrait;
 }
+
+// TODO: Move to a more appropriate location
```
Comment (karlprieb) [inline lib/pages/drive_detail/components/drive_explorer_item_tile.dart:361]: when?

File: lib/pages/drive_detail/drive_detail_page.dart [hunk 14]
```diff
@@ -906,3 +974,107 @@ class CustomBottomNavigation extends StatelessWidget {
     );
   }
 }
+
+// TODO: WIP!
```
Comment (karlprieb) [inline lib/pages/drive_detail/drive_detail_page.dart:978]: ?

File: lib/theme/theme.dart [hunk 15]
```diff
@@ -1,10 +1,23 @@
...
+    final brightness = WidgetsBinding.instance.window.platformBrightness;
+    if (brightness == Brightness.dark) {
+      return ArDriveThemes.dark;
+    } else {
+      return ArDriveThemes.light;
+    }
+  }
+}
+
+// TODO(@thiagocarvalhodev): remove this code.
```
Comment (karlprieb) [inline lib/theme/theme.dart:20]: when?

File: lib/utils/add_debounce.dart [hunk 16]
```diff
@@ -0,0 +1,5 @@
+import 'dart:async';
+
+addDebounce(Function() function) async {
```
Comment (karlprieb) [inline lib/utils/add_debounce.dart:3]: `debounce` is a more appropriate name

File: lib/authentication/ardrive_auth.dart (hunk 1 above)
Comment (thiagocarvalhodev) [reply lib/authentication/ardrive_auth.dart:22]: 0692fa13007b006b30e659c31fb460444a65538d

File: lib/components/ghost_fixer_form.dart (hunk 10 above)
Comment (thiagocarvalhodev) [reply lib/components/ghost_fixer_form.dart:52]: It's the same behavior of before, so I just kept.

File: lib/pages/drive_detail/drive_detail_page.dart (hunk 14 above)
Comment (thiagocarvalhodev) [reply lib/pages/drive_detail/drive_detail_page.dart:978]: The grid view is progress yet. It's functional but we won't ship... yet!

File: lib/components/ghost_fixer_form.dart (hunk 10 above)
Comment (karlprieb) [reply lib/components/ghost_fixer_form.dart:52]: Do we really need to pop twice?

**Pull request merged**